### DISCUSSION TITLE
QA template not rendering

### DISCUSSION THREAD

**Discussion by dpiro** [chhoumann/quickadd#579] [Q&A]
I apologize in advance for what seems like an incredibly straightforward issue - but I’m unable to resolve myself and appreciate your assistance.

I’m using QA to populate data into a template, which also formats data via the Templater plugin. The QA Template prompts for a Company name and Effective Date and Days, which are loaded into certain Properties in frontmatter. In short, none of the Templater functions are rendering when I activate via QA. Note when I create a note based on the template manually without QA, the Templater functions do render properly.

Here are examples from the template using QA fields and Templater functions.

```
---
effective: {{VALUE:dtEffective_YYYY-MM}}-{{VALUE:dayDD}}
company: {{VALUE:company}}
createdate: <% tp.date.now("dddd YYYY-MM-DD") %> 
alias: NAV<% tp.frontmatter.company %>-<% tp.date.now("YYYY-MM", 0, tp.frontmatter.effective, "YYYY-MM-DD") %>
tags: [{{VALUE:company}}, NAV]
---

dtReceived:: <% tp.date.now("YYYY-MM", 0, tp.frontmatter.createdate, "YYYY-MM-DD") %>

```
Here is the result:

```
---
effective: 2023-09-30
company: CoName
createdate: <% tp.date.now("dddd YYYY-MM-DD") %> 
createdate2: <% tp.file.creation_date("dddd YYYY-MM-DD") %> 
alias: NAV<% tp.frontmatter.company %>-<% tp.date.now("YYYY-MM", 0, tp.frontmatter.effective, "YYYY-MM-DD") %>
tags: [CoName, NAV]
---

dtReceived:: <% tp.date.now("YYYY-MM", 0, tp.frontmatter.createdate, "YYYY-MM-DD") %>
```

Why would the Templater functions not render when activated by QA?

Thank you -

**Comment by TimmyChannel** [comment]
Hello, how did you resolve this issue?

**Comment by chhoumann** [comment]
Coming back to this long after the fact - sorry. Since @TimmyChannel asked how it was resolved, here's the current state.

On recent QuickAdd versions, Templater *is* run on notes QuickAdd creates from a Template choice, so literal `<% ... %>` surviving into the note like in your paste shouldn't happen anymore. If you still see it on a current version, please post the exact template text plus any error from the developer console (Ctrl+Shift+I) and I'll dig in.

That said, for this particular template I'd drop Templater entirely - everything in it has a QuickAdd-native token, and one engine is simpler than two:

```
---
effective: {{VDATE:effective,YYYY-MM-DD}}
company: {{VALUE:company}}
createdate: {{DATE:dddd YYYY-MM-DD}}
alias: NAV{{VALUE:company}}-{{VDATE:effective,YYYY-MM}}
tags: [{{VALUE:company}}, NAV]
---

dtReceived:: {{DATE:YYYY-MM}}
```

Two tricks do the work here:

- Named variables prompt once and are reused everywhere: `{{VALUE:company}}` appears three times but asks once, which replaces `tp.frontmatter.company`.
- `{{VDATE:effective,...}}` asks for the date once (natural language like "last day of september" works) and re-renders it in a different format wherever you repeat the token - `{{VDATE:effective,YYYY-MM-DD}}` in `effective` and `{{VDATE:effective,YYYY-MM}}` in the alias share one prompt, which replaces the `tp.date.now(..., tp.frontmatter.effective, ...)` reformatting and also merges your separate `dtEffective_YYYY-MM` and `dayDD` prompts into one.

Docs: https://quickadd.obsidian.guide/docs/FormatSyntax
